### PR TITLE
Improve message for S2386 and S3887

### PR DESCRIPTION
**Pull request by csaba-sagi-sonarsource**

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.CSharp/Rules/MutableFieldsShouldNotBe.cs [hunk 1]
```diff
@@ -48,63 +45,44 @@ public abstract class MutableFieldsShouldNotBe : SonarDiagnosticAnalyzer
...
-                    return;
-                }
-
-                if (!(analysisContext.SemanticModel.GetDeclaredSymbol(fieldDeclaration.Declaration.Variables[0]) is IFieldSymbol fieldSymbol) ||
-                    fieldSymbol.Type == null ||
-                    fieldSymbol.GetEffectiveAccessibility() != Accessibility.Public ||
-                    IsImmutableOrValidMutableType(fieldSymbol.Type))
+                if (HasAllInvalidModifiers(fieldDeclaration)
+                    && fieldDeclaration.Declaration.Variables.Count > 0
+                    && analysisContext.SemanticModel.GetDeclaredSymbol(fieldDeclaration.Declaration.Variables[0]) is IFieldSymbol { Type: not null } fieldSymbol
```
Comment (csaba-sagi-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/Rules/MutableFieldsShouldNotBe.cs:66]: These two conditions are not covered, I could not come up even with broken syntax that would cover it.

File: analyzers/src/SonarAnalyzer.CSharp/Rules/MutableFieldsShouldNotBe.cs [hunk 2]
```diff
@@ -48,63 +45,44 @@ public abstract class MutableFieldsShouldNotBe : SonarDiagnosticAnalyzer
...
-                // and all initializations to this field are immutable
-                var incorrectFieldVariables = CollectInvalidFieldVariables(fieldDeclaration, assignmentsImmutability,
-                        analysisContext.SemanticModel)
-                    .ToSentence(quoteWords: true);
-
-                if (incorrectFieldVariables != null)
-                {
-                    analysisContext.ReportDiagnosticWhenActive(Diagnostic.Create(SupportedDiagnostics[0],
-                        fieldDeclaration.Declaration.Type.GetLocation(), incorrectFieldVariables));
+                    var pluralize = incorrectFieldVariables.Count > 1 ? "s" : string.Empty;
```
Comment (andrei-epure-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/Rules/MutableFieldsShouldNotBe.cs:73]: I suggest to rename to `pluralSuffix` (`pluralize` sounds like a method name or boolean flag)

File: analyzers/src/SonarAnalyzer.CSharp/Rules/MutableFieldsShouldNotBe.cs [hunk 3]
```diff
@@ -48,63 +45,44 @@ public abstract class MutableFieldsShouldNotBe : SonarDiagnosticAnalyzer
...
+                    analysisContext.ReportDiagnosticWhenActive(
+                        Diagnostic.Create(SupportedDiagnostics[0],
+                        fieldDeclaration.Declaration.Type.GetLocation(),
+                        pluralize, incorrectFieldVariables.ToSentence(quoteWords: true)));
```
Comment (andrei-epure-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/Rules/MutableFieldsShouldNotBe.cs:77]: should be indented, one arg per line
```suggestion
                    analysisContext.ReportDiagnosticWhenActive(
                        Diagnostic.Create(SupportedDiagnostics[0],
                                                     fieldDeclaration.Declaration.Type.GetLocation(),
                                                     pluralize, 
                                                     incorrectFieldVariables.ToSentence(quoteWords: true)));
```

**Review by andrei-epure-sonarsource — APPROVED** [review]
LGTM (minor cosmetic suggestions)

**Pull request merged**